unpackaged/vesuvio_calibration/tests/system/test_system_analysis.py

```diff
@@ -0,0 +1,272 @@
+
+        #  Specify detectors tolerances set by user, then update with those to mask as invalid.
+        detector_specific_r_tols = {"L1": {116: 0.65, 170: 0.75}}
+        detector_specific_r_tols["L1"].update({k: TestConstants.INVALID_DETECTOR for k in [138, 141, 146, 147, 156, 158, 160, 163, 164,
+                                                                                           165, 167, 168, 169, 170, 182, 191, 192]})
+        self._assert_parameters_match_expected(params_table, detector_specific_r_tols)
+        self._assert_E1_parameters_match_expected(params_table, 3e-3, "Both")
+
+    def _assert_theta_parameters_match_expected(self, params_table, rel_tolerance):
+        thetas = params_table.column('theta')
+        actual_thetas = self._calibrated_params['theta']
+
+        self.assertFalse(np.isnan(thetas).any())
+        self.assertFalse(np.isinf(thetas).any())
+        return assert_allclose_excluding_bad_detectors(actual_thetas, thetas, rel_tolerance)
+
+    def _assert_L1_parameters_match_expected(self, params_table, rel_tolerance):
+        L1 = params_table.column('L1')
+        actual_L1 = self._calibrated_params['L1']
+
+        #  Filter invalid detectors then mask
+        invalid_detectors = [(k, rel_tolerance.pop(k))[0] for k, v in copy(rel_tolerance).items() if v == TestConstants.INVALID_DETECTOR]
+        invalid_detector_mask = np.zeros(len(L1))
+        for index in invalid_detectors:
+            invalid_detector_mask[index] = 1
+
+        self.assertFalse(np.isnan(np.ma.masked_array(L1, mask=invalid_detector_mask)).any())
+        self.assertFalse(np.isinf(L1).any())
+        return assert_allclose_excluding_bad_detectors(np.ma.masked_array(actual_L1, mask=invalid_detector_mask),
+                                                       np.ma.masked_array(L1, mask=invalid_detector_mask), rel_tolerance)
+
+    def _assert_E1_parameters_match_expected(self, params_table, rel_tolerance, fit_type):
+        if fit_type != "Shared":
+            E1 = params_table.column('E1')[0]
+            self.assertAlmostEqual(E1, EVSGlobals.ENERGY_ESTIMATE, delta=EVSGlobals.ENERGY_ESTIMATE*rel_tolerance)
+
+        if fit_type != "Individual":
+            global_E1 = params_table.column('global_E1')[0]
+            self.assertAlmostEqual(global_E1, EVSGlobals.ENERGY_ESTIMATE, delta=EVSGlobals.ENERGY_ESTIMATE*rel_tolerance)
+
+    def _assert_parameters_match_expected(self, params_table, tolerances=None):
+        rel_tol_theta, rel_tol_L1 = self._extract_tolerances(tolerances)
+        theta_errors = self._assert_theta_parameters_match_expected(params_table, rel_tol_theta)
+        L1_errors = self._assert_L1_parameters_match_expected(params_table, rel_tol_L1)
+
+        if theta_errors or L1_errors:
+            raise AssertionError(f"Theta: {theta_errors})\n L1: {L1_errors}")
+
+    @staticmethod
+    def _extract_tolerances(tolerances: dict) -> (dict, dict):
```

Comment:
Should we remove the type annotation to maintain consistency with the other functions in the file? what do you think?